src/dlnodesession.cpp

```diff
@@ -0,0 +1,294 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "dlnodesession.hpp"
+
+#include <map>
+#include <string>
+
+#include "logging.hpp"
+#include "modelinstance.hpp"
+#include "modelinstanceunloadguard.hpp"
+#include "modelmanager.hpp"
+#include "nodeinputhandler.hpp"
+#include "nodeoutputhandler.hpp"
+#include "nodestreamidguard.hpp"
+#include "tensorinfo.hpp"
+
+namespace ovms {
+DLNodeSession::DLNodeSession(const NodeSessionMetadata& metadata, const std::string& nodeName, uint32_t inputsCount, const ModelManager& manager, const std::string& modelName, model_version_t modelVersion) :
+    NodeSession(metadata, nodeName, inputsCount),
+    modelManager(manager),
+    modelName(modelName),
+    modelVersion(modelVersion) {}
+
+DLNodeSession::DLNodeSession(const NodeSessionMetadata&& metadata, const std::string& nodeName, uint32_t inputsCount, const ModelManager& manager, const std::string& modelName, model_version_t modelVersion) :
+    NodeSession(std::move(metadata), nodeName, inputsCount),
+    modelManager(manager),
+    modelName(modelName),
+    modelVersion(modelVersion) {}
+
+DLNodeSession::~DLNodeSession() = default;
+
+void DLNodeSession::clearInputs() {
+    this->inputHandler->clearInputs();
```

Comment:
Why do we need this method? Session is not reusable, from my understanding this is per node activation.

Reply:
Because we do not want to keep the inputs longer than necessary. That was the purpose for having clearInputs in dl_node.cpp as well.